### PR TITLE
Improve performance

### PR DESCRIPTION
**Pull request by mohd-akram**
While profiling, I noticed `getCode()` was taking too much time. These changes improve performance. To maintain the API, we still return copies via `slice()` and `Object.assign()`. Performance can be improved more by removing those.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,44 +1,56 @@
...
-      return (
-        language.name.toLowerCase() === name.toLowerCase() ||
-        language.nativeName.toLowerCase() === name.toLowerCase()
-      );
-    });
-    return code || '';
+    name = name.toLowerCase();
+    return LANGUAGES_BY_NAME.hasOwnProperty(name)
+      ? LANGUAGES_BY_NAME[name].code
+      : '';
```
Comment (meikidd) [inline src/index.js:49]: The original code also supports getting code by native name. So in the new code, should also check `LANGUAGE_NATIVE_NAMES.hasOwnProperty(name)`

**Comment by meikidd** [comment]
@mohd-akram  Thank you for your neat work!

File: src/index.js (hunk 1 above)
Comment (mohd-akram) [reply src/index.js:49]: This case is handled as well. When `LANGUAGES_BY_NAME` is constructed I use the native name as key as well so we handle both cases and save an extra lookup.

**Comment by meikidd** [comment]
@mohd-akram New version v3.1.0 released, please help to check

**Comment by mohd-akram** [comment]
@meikidd Thanks! Works well.

**Pull request merged**